### DISCUSSION TITLE
Info on Hiro wallet login implementation in a standalone videogame

### DISCUSSION THREAD

**Discussion by Ghislo749** [blockstack/stacks-blockchain-api#1577] [Q&A]
Hello people! I'm not sure I'm posting this in the right place, eventually you can let me know and I will remove it, or move it to the right place..

I'm developing a little videogame and I implemented a wallet authentication for EVM chains that lets the users login with their wallet. The way I did it for MetaMask is through [chainsafe](https://chainsafe.io) service. This is because the game is not Web3 or WebGL, but it's developed as Standalone and I need the game to communicate with the wallet through the browser, to make sure the user logging in is the owner of the wallet. 

I share with you a little video that shows how it works for MetaMask and Ethereum mainnet. As you can see, chainsafe generates a code after signing the authentication, that is copied and pasted into the game to confirm ownership of the wallet. The wallet is not connected to the game, it's only used to make sure you login with an NFT you actually own, so it's a one time authentication.

**I want to give the users the possibility to login in the same way, but using Hiro wallet on Stacks blockchain**, but I can't find anything similar to the chainsafe service and I was wondering if someone from Hiro's team might have an idea on how I could implement it.

After confirming ownership the game makes an API call and retrieves data of NFTs owned by that wallet and uses it to populate the game and eventually the game database.

After login the user can choose an NFT to use as login into the game. All the progress in the game will be linked to the chosen NFT.

I think such a service could be a great implementation for other developers as well.

Here's my discord: Ghislo749#5047

https://user-images.githubusercontent.com/17834556/224515847-846cbc0d-b125-46fe-a7b2-7972d51bb963.mp4

**Comment by friedger** [comment]
Have you looked into wallet connect? Or is there no internet connection?

**Reply by Ghislo749** [reply]
Yeah I already implemented Wallect Connect for EVM chains, it's not shown in the video, but does it work with Hiro wallet? I don't see it on their website.. 

https://explorer.walletconnect.com/?search=Hiro

**Comment by friedger** [comment]
https://docs.xverse.app/wallet-connect/web/webapp
